tests/phpunit/tests/query/cacheResults.php

```diff
@@ -302,31 +370,51 @@ public function data_query_cache_duplicate() {
 					'post_type'      => 'post',
 				),
 			),
-			'same args'                 => array(
+			'same args'                   => array(
 				'query_vars1' => array( 'post_type' => 'post' ),
 				'query_vars2' => array( 'post_type' => 'post' ),
 			),
-			'same args any'             => array(
+			'same args any'               => array(
 				'query_vars1' => array( 'post_type' => 'any' ),
 				'query_vars2' => array( 'post_type' => 'any' ),
 			),
-			'any and post types'        => array(
+			'any and post types'          => array(
 				'query_vars1' => array( 'post_type' => 'any' ),
 				'query_vars2' => array( 'post_type' => array( 'post', 'page', 'attachment' ) ),
 			),
-			'different order post type' => array(
+			'different order post type'   => array(
 				'query_vars1' => array( 'post_type' => array( 'post', 'page' ) ),
 				'query_vars2' => array( 'post_type' => array( 'page', 'post' ) ),
 			),
-			'post status array'         => array(
+			'post status array'           => array(
 				'query_vars1' => array( 'post_status' => 'publish' ),
 				'query_vars2' => array( 'post_status' => array( 'publish' ) ),
 			),
-			'post status order'         => array(
+			'post status order'           => array(
 				'query_vars1' => array( 'post_status' => array( 'draft', 'publish' ) ),
 				'query_vars2' => array( 'post_status' => array( 'publish', 'draft' ) ),
 			),
-			'cache parameters'          => array(
+			'post in order'               => array(
+				'query_vars1' => array( 'post__in' => array( 1, 2, 3, 4, 5 ) ),
+				'query_vars2' => array( 'post__in' => array( 5, 4, 3, 2, 1 ) ),
+			),
+			'post parent in order'        => array(
+				'query_vars1' => array( 'post_parent__in' => array( 1, 2, 3, 4, 5 ) ),
+				'query_vars2' => array( 'post_parent__in' => array( 5, 4, 3, 2, 1 ) ),
+			),
+			'post name in order'          => array(
+				'query_vars1' => array( 'post_name__in' => array( 'elphaba', 'glinda', 'the-wizard-of-oz', 'doctor-dillamond' ) ),
+				'query_vars2' => array( 'post_name__in' => array( 'doctor-dillamond', 'elphaba', 'the-wizard-of-oz', 'glinda' ) ),
+			),
+			'term queries order (array)'  => array(
+				'query_vars_1' => array( 'cat' => array( '1', '2' ) ),
+				'query_vars_2' => array( 'cat' => array( '2', '1' ) ),
+			),
+			'term queries order (string)' => array(
+				'query_vars_1' => array( 'cat' => '1,2' ),
+				'query_vars_2' => array( 'cat' => '2,1' ),
+			),
```

Comment:
These two are failing. `cat` is normalised in the generated SQL query but not in the arguments passed to `::generate_cache_key()`.

I can't figure out why as the query vars are passed to `::parse_term_query()` by reference so sorting it there should work. @joemcgill Can you see what I am missing?

Reply:
> These two are failing. `cat` is normalised in the generated SQL query but not in the arguments passed to `::generate_cache_key()`.
> 
> I can't figure out why as the query vars are passed to `::parse_term_query()` by reference so sorting it there should work. @joemcgill Can you see what I am missing?

nvm, the tests were testing the wrong thing so I've started testing the cache key generated by WP_Query in bc246274c3ceb7a3204ea113216919177368d9e4